pkg/controller/notification/notification_controller.go

```diff
@@ -100,24 +112,37 @@ func (r *ReconcileNotification) Reconcile(request reconcile.Request) (reconcile.
 		}, nil
 	}
 
-	reqLogger.Info("Notification has been sent")
-	err = r.setStatusNotificationSent(notification)
+	// Send the notification - first create the notification context
+	notCtx, err := NewNotificationContext(r.client, notification.Spec.UserID, request.Namespace, r.config)
 	if err != nil {
-		reqLogger.Error(err, "unable to set sent status to Notification")
-		return reconcile.Result{}, err
+		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, notification,
+			r.setStatusNotificationContextError, err, "failed to create notification context")
+	}
+
+	// if the environment is set to e2e do not attempt sending via mailgun
+	if r.config.GetEnvironment() != "e2e-tests" {
```

Comment:
Let's log an info message in case of `e2e-tests` saying that the notifications was skipped.

Reply:
@alexeykazakov good idea, i've added it here: 09233c4dd5f3c2ed1b15defbc78c7f11eca29639